### ISSUE TITLE
Add accessible content to John Travoltage

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/john-travoltage#89]
@jobara and @jhung are ready to implement their design for accessible content into the John Travoltage simulation.  This issue will track related commits and issues for this project.

**Comment by jobara** [comment]
@jessegreenberg last week when we were pairing it appeared that the range input (slider) we added to the parallel DOM wasn't working. I looked into this a bit more last week and it actually appears that it is working. The problem is that the developer tools don't show "user" directed change on the input. However, adding an event listener for the `input` event captures the changes.

**Comment by jessegreenberg** [comment]
@jobara, that is great! It is also good to know this detail about the developer tools.  Thanks for letting me know.

**Comment by jobara** [comment]
The sliders `< input type="range" >` were not keyboard accessible in Safari when run in the sim. It seems the issue was that in Safari the sliders require some width. Simply adding a width of 1px; seems to have done the trick.

**Comment by jessegreenberg** [comment]
Ah, good catch @jobara.  Thanks for figuring this out, that is good to know!

**Comment by jobara** [comment]
It seems that VoiceOver is reading the "aria-valuetext" for the sliders twice. So we get things like "Position 16: foot is on the carpet. Position16: foot is on the carpet"

**Comment by jobara** [comment]
Another issue is that the alerts for the discharge are not being announced. Steve Faulkner wrote up 4 methods for adding an alert. https://www.paciellogroup.com/blog/2012/06/html5-accessibility-chops-aria-rolealert-browser-support/

## results ##

Below are the results of testing his demo page across screen readers and browsers.
http://www.html5accessibility.com/tests/alert-test.html

### JAWS ###

Browser | Method 1 | Method 2 | Method 3 | Method 4
------------|--------------|--------------|---------------|------------
Chrome | yes | yes | yes | yes 
Firefox | yes | yes | yes | yes
IE 11 | yes | yes | yes | yes

### NVDA ###

Browser | Method 1 | Method 2 | Method 3 | Method 4
------------|--------------|--------------|---------------|------------
Chrome | yes | yes | yes | yes
Firefox | yes | yes | yes | yes
IE 11 | no | no | yes | yes 

### VoiceOver ###

Browser | Method 1 | Method 2 | Method 3 | Method 4
------------|--------------|--------------|---------------|------------
Chrome | no | no | no | no
Firefox | no | no | no | no
Safari | yes | yes | yes | yes

**Comment by terracoda** [comment]
@jobara, did you fix the duplicate content that Safari is reading out? We had a lot of duplicate in our initial interviews in other screen readers as well.

**Comment by jobara** [comment]
@terracoda not yet. I've started to look at the alerts first and will come back to the duplicate issue after.

**Comment by jobara** [comment]
In regards to the alert. I'm having better results by adding and removing an element from the DOM as the alert is needed. In this case I always add a `<div>` and add and remove a `<p role="alert">` to it, as needed.

**Comment by jessegreenberg** [comment]
From https://github.com/phetsims/john-travoltage/issues/89#issuecomment-220352858, I ran in to the same problem with checkboxes. Adding a pixel to the width added the check box to the navigation order.  Thanks @jobara!

**Comment by jobara** [comment]
@jessegreenberg do you know why the parallel DOM is positioned at the top left corner and clipped? https://github.com/phetsims/scenery/blob/master/js/accessibility/AccessibleInstance.js#L57-L66

I was testing on iOS today and noticed that I couldn't drag my finger around the screen to discover what elements were present. I'm thinking to instead have the accessibility element take 100% of the page and be layered on top of the sim. Then we could create large target zones to interact with the DOM elements. However, I don't know if there was a specific reason for trying to hide the accessibility elements.

**Comment by jobara** [comment]
It seems that IE 11 doesn't support `role="status"`. However, it appears that explicitly adding the implicit aria attributes `aria-live="polite"` and `aria-atomic=true` enables IE 11 to report the live content. I will need to test this across browser/AT combinations.

**Comment by jobara** [comment]
Test results when using `role="status" aria-live="polite" aria-atomic=true`

Browser | JAWS      | NVDA      | VoiceOver |
------------|--------------|--------------|----------------
Chrome | yes | yes | yes 
Firefox | yes | yes | no
IE 11 | yes | yes | N/A
Safari | N/A | N/A | yes

The only one it doesn't work for is FF with VoiceOver,

**Comment by jobara** [comment]
@jessegreenberg could you take a look at this. I hoping it can be merged into master and for this issue to be closed. There are still improvements and etc that @jhung and I plan to file specific issues for once this is in.

**Comment by jessegreenberg** [comment]
Great @jobara, thanks! Sure will take a look at this today.

**Comment by jessegreenberg** [comment]
@jobara since code review is complete, I opened #106 to track the merge.  Feel free to close this issue when you are ready.

**Comment by jobara** [comment]
@jessegreenberg great thanks.

**Comment by jobara** [comment]
Closing this issue as the Code Review ( https://github.com/phetsims/john-travoltage/issues/94 ) has been complete. Any more work on John Travoltage a11y should be tracked under another issue.
